### PR TITLE
Support customized onnx op 

### PR DESCRIPTION
**Pull request by mzmssg**
Sub feature for #227.
ONNX op with "com.microsoft.nnfusion.custom" domain would be directly matched to a NNFusion generic op.

image

### CODE REVIEW

File: src/nnfusion/frontend/onnx_import/op/custom_op.hpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+namespace nnfusion
+{
+    namespace frontend
+    {
+        namespace onnx_import
+        {
+            namespace set_1
+            {
+                NamedNodeVector TranslateCustomrOp(const onnx::NodeProto& node_proto,
```
Comment (xysmlx) [inline src/nnfusion/frontend/onnx_import/op/custom_op.hpp:36]: Customr -> Custom?

**Comment by xysmlx** [comment]
Shall we convert other unknown ONNX ops (e.g., unsupported standard ONNX ops) to custom ops?

**Comment by mzmssg** [comment]
> Shall we convert other unknown ONNX ops (e.g., unsupported standard ONNX ops) to custom ops?

Converting all unkonw op(even not supported by backend) might mess the error message. So I prefer to convert known ops and report unsupport as early as possible.
Of course, once backend ready, the convert func could be reused by one-line register `REGISTER_OPERATOR("XX_OP", 1, TranslateCustomOp);`

**Comment by xysmlx** [comment]
> > Shall we convert other unknown ONNX ops (e.g., unsupported standard ONNX ops) to custom ops?
> 
> Converting all unkonw op(even not supported by backend) might mess the error message. So I prefer to convert known ops and report unsupport as early as possible.
> Of course, once backend ready, the convert func could be reused by one-line register `REGISTER_OPERATOR("XX_OP", 1, TranslateCustomOp);`

I agree with your concerns. But in this way, users need to re-build the nnfusion after the modification. Maybe we could enable the custom converter when users set a flag.

**Comment by mzmssg** [comment]
> > > Shall we convert other unknown ONNX ops (e.g., unsupported standard ONNX ops) to custom ops?
> > 
> > 
> > Converting all unkonw op(even not supported by backend) might mess the error message. So I prefer to convert known ops and report unsupport as early as possible.
> > Of course, once backend ready, the convert func could be reused by one-line register `REGISTER_OPERATOR("XX_OP", 1, TranslateCustomOp);`
> 
> I agree with your concerns. But in this way, users need to re-build the nnfusion after the modification. Maybe we could enable the custom converter when users set a flag.
> 
> BTW, the pipeline failed due to code format.

In this case, users should be aware that an customized op is used (by setting domain of "com.microsoft.nnfusion.custom") to avoid change & rebuild nnfusion. Consider this case: an op is introduced in op_set 11, but an op_set10 model use a namesake op. In ONNX, this model is illegal, but nnfusion sucessfully convert it to an op_set 11 op, isn't that confusing? Keep strict & recognize unknown ops might be better.

**Review by xysmlx — APPROVED** [review]

**Pull request merged**